dev/tasks/r/github.packages.yml

```diff
@@ -375,29 +375,30 @@ jobs:
           read_parquet(system.file("v0.7.1.parquet", package = "arrow"))
           print(arrow_info())
 
-  #TODO test macos source build?
-  test-linux-source:
+  test-source:
     needs: source
-    name: Test linux source build
-    runs-on: ubuntu-latest
+    name: Test {{ '${{ matrix.platform.name }}' }} source build
+    runs-on: {{ '${{ matrix.platform.runs_on }}' }}
+    strategy:
+      fail-fast: false
+      matrix:
+        platform:
+          - {runs_on: "ubuntu-latest", name: "Linux"}
+          - {runs_on: ["self-hosted", "macos-10.13"] , name: "macOS"}
     steps:
       - name: Install R
+        if: matrix.platform.name == 'Linux'
         uses: r-lib/actions/setup-r@v2
-        with:
-          install-r: false
       {{ macros.github_setup_local_r_repo(false, false)|indent }}
-      {{ macros.github_checkout_arrow()|indent }}
+      {{ macros.github_checkout_arrow(action_v="3")|indent }}
       - name: Install sccache
+        if: matrix.platform.name == 'Linux'
         shell: bash
         run: |
           arrow/ci/scripts/install_sccache.sh unknown-linux-musl /usr/local/bin
-      - name: Install R package system dependencies
-        run: |
-          sudo arrow/ci/scripts/r_install_system_dependencies.sh
-        env:
-          ARROW_GCS: "ON"
-          ARROW_S3: "ON"
-          ARROW_SOURCE_HOME: arrow
+      - name: Install R package system dependencies (Linux)
+        if: matrix.platform.name == 'Linux'
+        run: sudo apt-get install -y libcurl4-openssl-dev libssl-dev

```

Comment:
Why do we want to use `apt-get install` directly instead of using `arrow/ci/scripts/r_install_system_dependencies.sh`?

Reply:
We don't need test bench etc. for this build so this is clearer and directly visible what gets installed.